spark_llm/spark_llm_assistant.py

```diff
@@ -234,6 +237,26 @@ def plot_df(self, df: DataFrame) -> None:
         code = response.content.replace("```python", "```").split("```")[1]
         exec(code)
 
+#     def udf_llm(self, header: str, desc: str):
+#         code = self._udf_chain.run(
+#             desc=desc
+#         )
+#         # reformat, add indent for each line
+#         code_split = code.split("\n")
+#         new_code = "\n\t".join(code_split)
+        
+#         udf = f"""def {header}:\n\t{new_code}"""
+#         self.log(f"UDF:\n{udf}")
+
+    def udf_llm(self, desc: str):
```

Comment:
How about let's make it a decorator. Similar to the current `@udf` API in PySpark. For example
```
@assistant.udf(returnType="int")
def my_udf(x: int):
    """Description of the udf"
    ...
```

Reply:
Yeah. Note that:
1. the function body is empty. It requires the doc string and the signature. 
2. we should still find a way to cache the generated UDF.